packages/core/src/modules/oob/OutOfBandApi.ts

```diff
@@ -483,7 +486,7 @@ export class OutOfBandApi {
         } else {
           // Wait until the connection is ready and then pass the messages to the agent for further processing
           this.connectionsApi
-            .returnWhenIsConnected(connectionRecord.id)
+            .returnWhenIsConnected(connectionRecord.id, config.timeoutMs ? { timeoutMs: config.timeoutMs } : undefined)
```

Comment:
I think we can simplify to this?

```suggestion
            .returnWhenIsConnected(connectionRecord.id, { timeoutMs: config?.timeoutMs })
```